fastly/notifications.go

```diff
@@ -0,0 +1,311 @@
+	Sort       string `json:"sort"`
+	Total      int    `json:"total"`
+	Type       string `json:"type"`
+}
+
+// SearchIntegrations retrieves filtered, paginated integrations.
+func (c *Client) SearchIntegrations(i *SearchIntegrationsInput) (*SearchIntegrationsResponse, error) {
+	p := "/notifications/integrations"
+
+	ro := &RequestOptions{
+		Params: map[string]string{},
+	}
+	if i.Cursor != nil {
+		ro.Params["cursor"] = *i.Cursor
+	}
+	if i.Limit != nil {
+		ro.Params["limit"] = strconv.Itoa(*i.Limit)
+	}
+	if i.Sort != nil {
+		ro.Params["sort"] = *i.Sort
+	}
+	if i.Type != nil {
+		ro.Params["type"] = *i.Type
+	}
+
+	resp, err := c.Get(p, ro)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var sir *SearchIntegrationsResponse
+	if err := json.NewDecoder(resp.Body).Decode(&sir); err != nil {
+		return nil, err
+	}
+
+	return sir, nil
+}
+
+// CreateIntegrationInput is used as input to the CreateIntegration function.
+type CreateIntegrationInput struct {
+	// Config is configuration specific to the integration type.
+	Config map[string]string
+	// Description is the user submitted description of the integration.
+	Description *string
+	// Name is the user submitted name of the integration.
+	Name *string
+	// Type is the type of integration
+	Type *string
+}
+
+// CreateIntegrationResponse is the response for creating a new integration.
+type CreateIntegrationResponse struct {
+	// ID of created integration.
+	ID *string `json:"integration_id"`
+}
+
+// CreateIntegration creates a new integration.
+func (c *Client) CreateIntegration(i *CreateIntegrationInput) (*CreateIntegrationResponse, error) {
+	resp, err := c.PostJSON("/notifications/integrations", i, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var cir *CreateIntegrationResponse
+	if err := json.NewDecoder(resp.Body).Decode(&cir); err != nil {
+		return nil, err
+	}
+	return cir, nil
+}
+
+// GetIntegrationInput is used as input to the GetIntegration function.
+type GetIntegrationInput struct {
+	// ID of integration to fetch (required).
+	ID *string
+}
+
+// GetIntegration retrieves a specified integration.
+func (c *Client) GetIntegration(i *GetIntegrationInput) (*Integration, error) {
+	if i.ID == nil {
+		return nil, ErrMissingID
+	}
+
+	path := fmt.Sprintf("/notifications/integrations/%s", *i.ID)
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var integration *Integration
+	if err := json.NewDecoder(resp.Body).Decode(&integration); err != nil {
+		return nil, err
+	}
+
+	return integration, nil
+}
+
+// UpdateIntegrationInput is used as input to the UpdateIntegration function.
+type UpdateIntegrationInput struct {
+	// Config is configuration specific to the integration type.
+	Config map[string]string
+	// Description is the user submitted description of the integration.
+	Description *string
+	// ID of integration to update (required).
+	ID *string
```

Comment:
Typically anything `(required)` (which means it's used as part of the API request path) I've not made a pointer because it's not optional (i.e. we _need_ that input in order to actually make the API request).

Reply:
Got it, I've aligned the PR in this respect.